core/src/types/error.rs

```diff
@@ -414,6 +414,13 @@ impl From<Error> for io::Error {
     }
 }
 
+#[cfg(feature = "layers-prometheus")]
```

Comment:
Hi, please don't add `From` for `Error`. Implement a `parse_prometheus_error` instead.